var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -284,9 +308,13 @@ def setup_build_environment(self, env):
         env.set('BUILD_BZIP2', 0)
         env.set('BZIP2_INCLUDE', spec['bzip2'].prefix.include)
         env.set('BZIP2_LIB', spec['bzip2'].libs.directories[0])
-        env.set('BUILD_ZLIB', 0)
-        env.set('ZLIB_INCLUDE', spec['zlib'].prefix.include)
-        env.set('ZLIB_LIB', spec['zlib'].libs.directories[0])
+        # The test suite of the End of life releases fails with our zlib
+        if self.version >= Version('5.30.0'):
```

Comment:
Do you know why? It'd be better to have non-vendored dependencies if anything links to libperl (if it exists) and also requires zlib.

Reply:
If zlib follows semver they haven't had a breaking change since 2002...

Reply:
Also: can it be that the test itself is broken?

Reply:
> If zlib follows semver they haven't had a breaking change since 2002...

AFAICS, semver is a much newer idea/propsal only used by relatively new software.

As you will see, zlib didn't "follow" semver, not in 2002. It didn't exist back than or if it did't it was not something that was known. I worked at the headqarters of the 2nd-largest Linux distributor at that time, did a lot of in-house software development for that Linux distribution too and we would have known about if semver.org were popular.

The old dominant projects (and espcially huge complex projects like the Linux kernel, gcc, clang, perl, openssl, openssh, GNOME, KDE etc.) rather follow their own model which they established many many years ago, maybe see this:
https://en.wikipedia.org/wiki/Software_versioning#Change_significance
https://gist.github.com/jashkenas/cbd2b088e20279ae2c8e
The version practice of thems seem to be rather based on the basic idea this guy describes as "Romantic versioning":
https://github.com/romversioning/romver

While the idea of having a strict rule like semver in place is in theory good, projects like large Linux have discovered, that they always had to adapt their numbering to what was most bearable and readable. For example: When the version numbers simply became too large because of many revisions then they changed the numbering scheme.

Please don't tell me that they are wrong, for not using semver. It's good if you use if for your projects and nice if a whole domain of project in some newer language do that, but the writers of semver.org have no autority to dictate their concept to everyone.

No, it's not "one" test, and yest, sometimes some individual tests are broken, but when it's many many, then in't not that many:
```py
# Looks like you failed 5 tests of 317.
     4412    #   Failed test at t/cz-03zlib-v1.t line 754.
     4413    #   Failed test at t/cz-03zlib-v1.t line 766.
     4414    #   Failed test at t/cz-03zlib-v1.t line 788.
     4415    #   Failed test at t/cz-03zlib-v1.t line 790.
# Looks like you failed 5 tests of 471
          got: '74154'
     ...
     4458    #     expected: '74154'
     4459    # Inflation Error: data error
     4460    #   Failed test 'got expected uncompressed data'
     4461    #   at t/cz-06gzsetp.t line 140.
     4462    # unc len = 247, input len = 74154

Looks like you failed 15 tests of 51.
```
- Especially a wrong test would not show data corruption, but fail because of some other issue.

I've tested and pushed the fix according to this link:

It's an issue which was fixed in 5.26: https://rt.cpan.org/Public/Bug/Display.html?id=120134
```py
-    depends_on('zlib+shared')
+    # :5.24.1 needs zlib@:1.2.8: https://rt.cpan.org/Public/Bug/Display.html?id=120134
+    depends_on('zlib+shared',        when='@5.26.2:')
+    depends_on('zlib+shared@:1.2.8', when='@5.20.3:5.24.1')
```